addons/mass_mailing/views/mailing_mailing_views.xml

```diff
@@ -159,7 +159,7 @@
                             <field name="mailing_type" widget="radio" options="{'horizontal': true}" invisible="1"
                                 attrs="{'readonly': [('state', 'in', ('sending', 'done'))]}"/>
                             <field name="subject" string="Subject" attrs="{'readonly': [('state', 'in', ('sending', 'done'))]}" widget="char_emojis" placeholder="e.g. New Sale on all T-shirts"/>
-                            <field name="preview" string="Preview Text" attrs="{'readonly': [('state', 'in', ('sending', 'done'))]}" widget="char_emojis" placeholder="e.g. Check it out before it's too late!"/>
+                            <field name="preview" string="Preview Text" attrs="{'readonly': [('state', 'in', ('sending', 'done'))], 'invisible':[('mailing_type', '=', 'sms')]}" widget="char_emojis" placeholder="e.g. Check it out before it's too late!"/>
```

Comment:
Should be done by inheritance in mass_mailing_sms. You will have to make an xpath on position=attributes, and rewrite the whole attrs value (we still cannot update part of attrs, we have to rewrite it completely). Not very modular but hey, that's what we have currently :D .